pallets/auctions/src/lib.rs

```diff
@@ -0,0 +1,223 @@
+        /// Only owner of NFT can list for auction
+        NftNotOwned,
+        /// buy_it_price should be greater then start_price
+        AuctionPricingInvalid,
+        /// NFT has not been listed for auction
+        NFTNotListedForAuction,
+        /// NFT cannot be set to the new state
+        NFTStateInvalid
+    }
+
+    #[pallet::hooks]
+    impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {}
+
+    // Dispatchable functions allows users to interact with the pallet and invoke state changes.
+    // These functions materialize as "extrinsics", which are often compared to transactions.
+    // Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+    #[pallet::call]
+    impl<T: Config> Pallet<T> {
+        /// An example dispatchable that takes a singles value as a parameter, writes the value to
+        /// storage and emits an event. This function must be dispatched by a signed extrinsic.
+        #[pallet::weight(10_000 + T::DbWeight::get().writes(1))]
+        pub fn create_auction(
+            origin: OriginFor<T>,
+            nft_id: NFTId,
+            marketplace_id: MarketplaceId,
+            #[pallet::compact] start_block: T::BlockNumber,
+            #[pallet::compact] end_block: T::BlockNumber,
+            start_price: BalanceCaps<T>,
+            buy_it_price: Option<BalanceCaps<T>>,
+        ) -> DispatchResultWithPostInfo {
+            let creator = ensure_signed(origin)?;
+            let current_block = frame_system::Pallet::<T>::block_number();
+            // ensure the auction timeline is valid
+            ensure!(start_block < end_block, Error::<T>::AuctionTimelineInvalid);
+            // ensure start block > current block
+            ensure!(
+                start_block > current_block,
+                Error::<T>::AuctionTimelineInvalid
+            );
+            // ensure maximum auction duration is not exceeded
+            ensure!(
+                end_block.checked_sub(&start_block) <= Some(T::MaxAuctionDuration::get()),
+                Error::<T>::AuctionTimelineInvalid
+            );
+            // ensure minimum auction duration is valid
+            ensure!(
+                end_block.checked_sub(&start_block) >= Some(T::MinAuctionDuration::get()),
+                Error::<T>::AuctionTimelineInvalid
+            );
+            // ensure buffer period is valid
+            ensure!(
+                start_block.checked_sub(&current_block) >= Some(T::MinAuctionBuffer::get()),
+                Error::<T>::AuctionTimelineInvalid
+            );
+
+            // ensure the buy_it_price is greater than start_price
+            match buy_it_price {
+                Some(price) => ensure!(price > start_price, Error::<T>::AuctionPricingInvalid),
+                None => (),
+            }
+
+            // ensure the caller is the owner of NFT
+            ensure!(
+                T::NFTHandler::owner(nft_id) == Some(creator.clone()),
```

Comment:
Instead of calling `owner`, `is_listed_for_sale` and `is_in_transmission` call `get_nft`. This will decrease the amount of time we access our db and lower the weight value.